src/process_utility.c

```diff
@@ -2625,6 +2690,11 @@ process_add_constraint_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
 				case CONSTR_PRIMARY:
 				case CONSTR_FOREIGN:
 					break;
+				case CONSTR_CHECK:
+				{
+					validate_check_constraint(chunk, con);
```

Comment:
Foreign keys also need validation, is it performed elsewhere? Would be good to have a comment describing the difference.

Reply:
for foreign keys postgres does the validation for us since it queries through SPI and we enable inheritance for those specific queries so compressed data is also checked